src/compiler/utilities.ts

```diff
@@ -5278,7 +5278,7 @@ export function isPushOrUnshiftIdentifier(node: Identifier) {
  *
  * @internal
  */
-export function isParameterDeclaration(node: Declaration): boolean {
+export function isParameterDeclaration(node: Declaration): node is ParameterDeclaration {

```

Comment:
Did you forget to undo this maybe? I think despite the terribly misleading name, this function is not meant to return whether `node is ParameterDeclaration`.

Reply:
Yeah, totally! I could swear that I have reverted this... but I guess I didn't 😬 I fixed this just now.

Reply:
Yeah, this definitely is not a type guard. See also #52283.